client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -145,10 +208,19 @@ function mediaButton( editor ) {
 			return;
 		}
 
+		const dirtyImage = MediaStore.getDirtyImage();
+		let numOfImagesToUpdate = 0;
+
 		isVisualEditMode = ! editor.isHidden();
 
 		if ( isVisualEditMode ) {
```

Comment:
Great catch!

I fixed this partly in 1c7926cea05f25dbf19d47ace4fdc8e683bbd488. Now, image is updated. However, the caption is not resized as it should be. Will look into this.